### ISSUE TITLE
Updating Void from older glibc

### ISSUE THREAD

**Issue by alekseyt** [voidlinux/void-packages#12290]
### System

* xuname:  
```
dmesg: read kernel buffer failed: Function not implemented
Void 4.4.0-43-Microsoft x86_64 AuthenticAMD notuptodate r
```

(Void in WSL)

* package:  
 xbps-0.51_19

### Expected behavior

``xbps-install`` works after updating packages.

### Actual behavior

``xbps-install: /usr/lib/libc.so.6: version `GLIBC_2.26' not found (required by /usr/lib/libssl.so.44)``

### Steps to reproduce the behavior

The problem is explained in this forums post: https://forum.voidlinux.eu/t/void-in-wsl/4873 (ignore all WSL stuff)).

When XPBS on Void with glibc-2.25 is being updated, it doesn't pull newer glibc, but libssl and libcrypto links with it. When ``xbps-install -Su`` is executed is has the following packages list for update:

```
# xbps-install -Su
[*] Updating `https://repo.voidlinux.eu/current/x86_64-repodata' ...
x86_64-repodata: 1504KB [avg rate: 4276KB/s]

Name        Action    Version           New version            Download size
libcrypto42 install   -                 2.6.4_1                614KB
libssl44    install   -                 2.6.4_1                107KB
libxbps     update    0.51_18           0.51_19                106KB
xbps        update    0.51_18           0.51_19                81KB
```

After updating libssl and libcrypto (i believe the fact that XBPS update failed is irrelevant), consequent calls to ``xbps-install``, ``xbps-query`` and so on displays the following error:

```
xbps-install: /usr/lib/libc.so.6: version `GLIBC_2.26' not found (required by /usr/lib/libssl.so.44)
xbps-install: /usr/lib/libc.so.6: version `GLIBC_2.26' not found (required by /usr/lib/libcrypto.so.42)
```

To work this around i had to do ``xbps-install -u glibc`` before attempting update.

I believe the problem is somewhere around missing/wrong dependency on libc. If i look into ``/var/db/xbps/pkgdb-0.38.plist``, as far as i can see, XBPS ``run_depends`` on glibc-2.25, libcrypto and libssl as well. So i suspect it's a sort of packaging/dependencies problem. Shouldn't libssl/libcrypto depend on glibc-2.26?

I'm not sure if this is WSL problem. I've checked repodata on Void w/o WSL and XBPS/libssl/libcrypto depends looks the same (glibc-2.25 instead of glibc-2.26).

I can provide more info if needed.

**Comment by Gottox** [comment]
d@voidlinux/pkg-committers how about staticly linking xbps like we do with bash?

**Comment by Gottox** [comment]
@alekseyt for you to fix: use a staticly linked version from https://repo.voidlinux.eu/static/

**Comment by ebfe** [comment]
Bumping common/shlibs in the case of glibc updates would work too, or #4666

**Comment by alekseyt** [comment]
I would try static XBPS but unfortunately XBPS install is blocked by another issue, seemingly WSL-related (user namespaces not implemented or something). This image i'm using was not prepared by me, it's some image from dockerhub, i think it's this one: https://hub.docker.com/r/voidlinux/voidlinux/ .

Although i'm confident installing static XBPS would eliminate solibs errors, yes. I'll try to check this when WSL issue is resolved (probably not anytime soon).

**Comment by Gottox** [comment]
*cough* I think I should update something..

**Comment by asergi** [comment]
In the meantime, same thing as #6015?

**Comment by ebfe** [comment]
Yes
